cpp/src/arrow/ipc/adapter.cc

```diff
@@ -226,15 +252,17 @@ class RowBatchReader::Impl {
     // we can skip that buffer without reading from shared memory
     FieldMetadata field_meta = metadata_->field(field_index_++);
 
+    // extract null_bitmap which is common to all arrays
+    std::shared_ptr<Buffer> null_bitmap;
+    if (field_meta.null_count == 0) {
+      null_bitmap = nullptr;
```

Comment:
Ah sorry, I meant that the nullary ctor for `shared_ptr` sets nullptr (versus having to do so explicitly)